### PR TITLE
Add support for post process classes

### PR DESCRIPTION
**Pull request by javierav**
This PR implements TailwindMerge and its configuration. It has been implemented globally so that the cache it incorporates is shared and used by all ClassVariants instances.

### CODE REVIEW

**Comment by javierav** [comment]
@adrianthedev tailwind_merge-0.13.1 requires ruby version < 4.0, >= 3.1 Should we drop support for Ruby 3.0?

**Review by adrianthedev — APPROVED** [review]

**Comment by javierav** [comment]
I'm not entirely convinced by the proposal of this PR, since we are just integrating with TailwindMerge. I think it would be better to allow something more generic and let the developer decide. What do you think @adrianthedev?

```ruby
ClassVariants.configure do |config|
  config.process_classes_with do |classes|
     TailwindMerge::Merger.new.merge(classes)
  end
end
```

**Comment by adrianthedev** [comment]
Yeah. That could work as well.
We could expose this precise block in the initializer. Maybe commented out for those who don't want to add `TailwindMerge` as a dependecy.

**Comment by javierav** [comment]
@adrianthedev ready for review!

File: lib/generators/class_variants/install/templates/class_variants.rb.tt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# ClassVariants.configure do |config|
+#   # allow to post process classes with an external utility like TailwindMerger
+#   config.process_classes_with do |classes|
+#     TailwindMerge::Merger.new.merge(classes)
+#   end
+# end
```
Comment (adrianthedev) [inline lib/generators/class_variants/install/templates/class_variants.rb.tt:8]: I was thinking about generating the initializer uncommented and leave only the `process_classes_with` option commented out, but I think I like this approach better.
We are not configuring anything with it, we might as well not load it.
Good call!

**Review by adrianthedev — APPROVED** [review]
Awesoem work @javierav!

**Pull request merged**